### DISCUSSION TITLE
Consider introducing a simple way to get a `MemoryAllocator` from the .NET Memory manager types

### DISCUSSION THREAD

**Discussion by john-h-k** [SixLabors/ImageSharp#1240] [Ideas]
The `System.Buffers` namespace contains the `ArrayPool<T>` and `MemoryPool<T>` types which are effectively the standard library allocator types. Many custom allocators are derived from these 2 types. Currently, the `ImageSharp` allocators don't seem to have a way get one based off one of these types. We have `ArrayPoolMemoryAllocator`, which is great, but it doesn't seem to offer a way to create this from a custom `ArrayPool<T>`, which can be frustrating

**Comment by antonfirsov** [comment]
It's relatively easy to extend ArrayPoolMemoryAllocator,  just needs a new constructor. I'm fine reviewing a PR dealing with this. Please note that ArrayPoolMemoryAllocator actually uses 2 array pools with different configurations: one for small, one for large buffers.

For MemoryPool we need a new MemoryAllocator type, so it's more complicated, needs analysis + design work + discussions.
